Services/Concurrency/LoopSettings.cs

```diff
@@ -35,6 +34,25 @@ public void NewLoop()
             // Prioritize connections and registrations, so that devices connect as soon as possible
             this.SchedulableFetches = Math.Max(1, this.ratingConfig.RegistryOperationsPerMinute / 25);
             this.SchedulableRegistrations = Math.Max(1, this.ratingConfig.RegistryOperationsPerMinute / 10);
+        }
+    }
+
+    public class PropertiesLoopSettings
+    {
+        public const int MIN_LOOP_DURATION = 500;
+
+        private readonly IRateLimitingConfig ratingConfig;
+
+        public double SchedulableTaggings { get; set; }
+
+        public PropertiesLoopSettings(IRateLimitingConfig ratingConfig)
+        {
+            this.ratingConfig = ratingConfig;
+            this.NewLoop();
+        }
+
+        public void NewLoop()
+        {
             this.SchedulableTaggings = Math.Max(1, this.ratingConfig.RegistryOperationsPerMinute / 25);
```

Comment:
this seems incorrect: tagging a device is not a registry operation, so the formula should use `TwinWritesPerSecond` instead.

Reply:
please add test to ensure the formula is the intended one

Reply:
Sure, I was just using the existing logic. To make sure the new value is correct, can you explain the logic behind the division by 25 and 10? Is that the max allowable by the hub?

Reply:
`TwinWritesPerSecond` is the max number of operations allowed before the hub starts throttling. The division is slow down the the actor further, to leave room to other threads competing for the same resource. For instance, if 100 is the quota, and if the actor is trying to tag 100 devices per second, a simulated device X trying to update the twin would have to wait for all the devices to be tagged before the device twin X is written. I would divide by 2, to leave 50% of the quota to potential simulated devices trying to update their twin.